ui/index.d.ts

```diff
@@ -0,0 +1,108 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+
+import { Subscription } from 'azure-arm-resource/lib/subscription/models';
+import { ServiceClientCredentials } from 'ms-rest';
+import { AzureEnvironment } from 'ms-rest-azure';
+import { Uri, TreeDataProvider, Disposable, TreeItem, Event } from 'vscode';
+
+export declare class AzureExplorer implements TreeDataProvider<IAzureNode>, Disposable {
+    public static readonly subscriptionContextValue: string;
+    public onDidChangeTreeData: Event<IAzureNode>;
+    /**
+     * Azure Explorer
+     * @param resourceProvider Describes the resources to be displayed under subscription nodes
+     * @param loadMoreCommandId The command your extension will register for the 'Load More...' node
+     */
+    constructor(resourceProvider: IChildProvider, loadMoreCommandId: string);
+    public getTreeItem(node: IAzureNode): TreeItem;
+    public getChildren(node?: IAzureParentNode): Promise<IAzureNode[]>;
+    public refresh(node?: IAzureNode, clearCache?: boolean): void;
+    public loadMore(node: IAzureNode): Promise<void>;
+    public showNodePicker(expectedContextValue: string): Promise<IAzureNode>;
+    public dispose(): void;
+}
+
+/**
+ * The AzureExplorer returns instances of IAzureNode, which are wrappers for IAzureTreeItem with relevant context and functions from the explorer
+ */
+export interface IAzureNode<T extends IAzureTreeItem = IAzureTreeItem> {
```

Comment:
Can't find any documentation on the "extends IAzureTreeItem = IAzureTreeItem" notation - what does it mean?

Reply:
It basically just makes the generic type optional. I got tired of always writing `IAzureNode<ExampleTreeItem>`, especially because I don't always need to interact with the tree item.

By doing `= IAzureTreeItem` you can just write `IAzureNode` and it means the same thing as `IAzureNode<IAzureTreeItem>`